net2/ip.go

```diff
@@ -2,19 +2,33 @@ package net2
 
 import (
 	"net"
+	"os"
 
 	"github.com/dropbox/godropbox/errors"
 )
 
 // This returns the list of local ip addresses which other hosts can connect
 // to (NOTE: Loopback ip is ignored).
+// Also resolves Hostname to an address and adds it to the list too, so
+// IPs from /etc/hosts can work too.
 func GetLocalIPs() ([]*net.IP, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to lookup hostname")
+	}
+	// Resolves IP Address from Hostname, this way overrides in /etc/hosts
+	// can work too for IP resolution.
+	ipInfo, err := net.ResolveIPAddr("ip4", hostname)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to resolve ip")
+	}
+	ips := []*net.IP{&ipInfo.IP}
+
+	// TODO(zviad): Is rest of the code really necessary?
```

Comment:
it's probably best to keep the following since InterfaceAddrs returns ip v6 localhost as well.